service/controller/app/v1/resource/chart/desired.go

```diff
@@ -38,34 +38,35 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 			APIVersion: chartAPIVersion,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        cr.Spec.Name,
+			Name:        cr.GetName(),
```

Comment:
This was wrong. cr.Spec.Name is the name of the app being installed e.g. prometheus. We now use the same name as the app CR.

It is up to the component creating app CRs to ensure the name is unique.